command/src/request.rs

```diff
@@ -143,51 +134,14 @@ impl fmt::Display for WorkerRequest {
     }
 }
 
-pub fn read_requests_from_file(file: &mut File) -> Result<Vec<WorkerRequest>, RequestError> {
-    let mut acc = Vec::new();
-    let mut buffer = Buffer::with_capacity(200000);
-    loop {
-        let previous = buffer.available_data();
-
-        let bytes_read = file.read(buffer.space()).map_err(RequestError::FileError)?;
-
-        buffer.fill(bytes_read);
-
-        if buffer.available_data() == 0 {
-            trace!("read_requests_from_file: empty buffer");
-            break;
-        }
-
-        let mut offset = 0usize;
-        match parse_several_requests::<WorkerRequest>(buffer.data()) {
-            Ok((i, requests)) => {
-                if !i.is_empty() {
-                    trace!("read_requests_from_file: could not parse {} bytes", i.len());
-                    if previous == buffer.available_data() {
-                        break;
-                    }
-                }
-                offset = buffer.data().offset(i);
+pub fn read_initial_state_from_file(file: &mut File) -> Result<InitialState, RequestError> {
```

Comment:
This function could be generic and only take a reader in parameter, which will be useful to create initial from everything

Reply:
done!